azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/maven/MavenSettingHelper.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.tools.auth.maven;
+
+import com.microsoft.azure.tools.auth.exception.AzureLoginException;
+import com.microsoft.azure.tools.auth.exception.InvalidConfigurationException;
+import com.microsoft.azure.tools.auth.exception.MavenDecryptException;
+import com.microsoft.azure.tools.auth.model.MavenAuthConfiguration;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.settings.Server;
+import org.apache.maven.settings.building.SettingsProblem;
+import org.apache.maven.settings.crypto.DefaultSettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecrypter;
+import org.apache.maven.settings.crypto.SettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecryptionResult;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+import java.util.Objects;
+
+public class MavenSettingHelper {
+    /**
+     * Get auth configuration from serverId.
+     *
+     * @param session           the maven session
+     * @param settingsDecrypter the decrypter
+     * @param serverId          the server id
+     * @return the auth configuration
+     * @throws AzureLoginException when there are errors decrypting maven generated password
+     */
+    public static MavenAuthConfiguration getAuthConfigurationFromServer(MavenSession session, SettingsDecrypter settingsDecrypter, String serverId)
+            throws AzureLoginException {
```

Comment:
it seems this method did 2/3 tasks:
get server from session.
(get raw configuration from server)
build configuration from raw configuration.

so why not extract 2 methods:
`Xpp3Dom getServerRawConfigution(MavenSession session, String serverId)`
`MavenAuthConfiguration buildConfiguration(Xpp3Dom rawConfiguration)`

Reply:
will move this class out of this module in later PR.